bitbots_hcm/src/bitbots_hcm/hcm_dsd/decisions/decisions.py

```diff
@@ -87,11 +86,11 @@ def perform(self, reevaluate=False):
         # we check if the values are actually changing, since the joint_state controller will publish the same message
         # even if there is no connection anymore. But we don't want to go directly to hardware error if we just
         # have a small break, since this can happen often due to loose cabling
-        if self.last_msg is not None and self.blackboard.current_joint_state is not None \
-                and not self.last_msg.position == self.blackboard.current_joint_state.position \
+        if self.blackboard.previous_joint_state is not None and self.blackboard.current_joint_state is not None \
+                and not self.blackboard.previous_joint_state.effort == self.blackboard.current_joint_state.effort \
+                and not self.blackboard.previous_joint_state.position == self.blackboard.current_joint_state.position \
```

Comment:
```suggestion
                and (self.blackboard.previous_joint_state.effort != self.blackboard.current_joint_state.effort \
                or self.blackboard.previous_joint_state.position != self.blackboard.current_joint_state.position) \
```
To have a different message, it is sufficient if either the efforts or the positions are different. Also, `!=` is more readable than `not ==`.